test/core/derivatives/derivatives_test.py

```diff
@@ -46,26 +47,42 @@
     problem.make_id() for problem in CONVOLUTION_TRANSPOSED_FAIL_PROBLEMS
 ]
 
+RNN_PROBLEMS = make_test_problems(RNN_SETTINGS)
+RNN_IDS = [problem.make_id() for problem in RNN_PROBLEMS]
 
-@pytest.mark.parametrize("problem", NO_LOSS_PROBLEMS, ids=NO_LOSS_IDS)
-def test_jac_mat_prod(problem, V=3):
+
+# selecting only every second rnn problem (only cpu problems)
+# because torch is not able to calculate this function on CUDA
+@pytest.mark.parametrize(
+    "problem", NO_LOSS_PROBLEMS + RNN_PROBLEMS, ids=NO_LOSS_IDS + RNN_IDS
+)
+def test_jac_mat_prod(problem, request, V=3):
     """Test the Jacobian-matrix product.
 
     Args:
         problem (DerivativesProblem): Problem for derivative test.
+        request: pytest request of problem
         V (int): Number of vectorized Jacobian-vector products.
     """
     problem.set_up()
     mat = torch.rand(V, *problem.input_shape).to(problem.device)
 
     backpack_res = BackpackDerivatives(problem).jac_mat_prod(mat)
-    autograd_res = AutogradDerivatives(problem).jac_mat_prod(mat)
+    if all(string in request.node.callspec.id for string in ["RNN", "cuda"]):
+        # torch does not implement cuda double-backwards pass on RNNs and
+        # recommends this workaround
+        with torch.backends.cudnn.flags(enabled=False):
+            autograd_res = AutogradDerivatives(problem).jac_mat_prod(mat)
```

Comment:
Move to `AutogradDerivatives`

Reply:
I'm not sure what you mean by this comment. In all cases ``AutogradDerivatives`` is used.

Reply:
Do you mean that this code should be moved inside the ``AutogradDerivatives(problem).jac_mat_prod(mat)`` function?

Reply:
Done.